src/metabase/driver/hive_like.clj

```diff
@@ -0,0 +1,173 @@
+    :month-of-year (hsql/call :month (hx/->timestamp expr))
+    :quarter (hsql/call :add_months
+                        (hsql/call :trunc (hx/->timestamp expr) (hx/literal :year))
+                        (hx/* (hx/- (hsql/call :quarter (hx/->timestamp expr))
+                                    1)
+                              3))
+    :quarter-of-year (hsql/call :quarter (hx/->timestamp expr))
+    :year (hsql/call :year (hx/->timestamp expr))))
+
+(defn date-interval [unit amount]
+  (hsql/raw (format "(NOW() + INTERVAL '%d' %s)" (int amount) (name unit))))
+
+(defn string-length-fn [field-key]
+  (hsql/call :length field-key))
+
+;; ignore the schema when producing the identifier
+(defn qualified-name-components
+  "Return the pieces that represent a path to FIELD, of the form `[table-name parent-fields-name* field-name]`."
+  [{field-name :name, table-id :table_id, parent-id :parent_id}]
+  (conj (vec (if-let [parent (metabase.models.field/Field parent-id)]
+               (qualified-name-components parent)
+               (let [{table-name :name, schema :schema} (db/select-one ['Table :name :schema], :id table-id)]
+                 [table-name])))
+        field-name))
+
+(defn field->identifier [field]
+  (apply hsql/qualify (qualified-name-components field)))
+
+;; copied from the Presto driver, except using mysql quoting style
+(defn apply-page [honeysql-query {{:keys [items page]} :page}]
+  (let [offset (* (dec page) items)]
+    (if (zero? offset)
+      ;; if there's no offset we can simply use limit
+      (h/limit honeysql-query items)
+      ;; if we need to do an offset we have to do nesting to generate a row number and where on that
+      (let [over-clause (format "row_number() OVER (%s)"
+                                (first (hsql/format (select-keys honeysql-query [:order-by])
+                                                    :allow-dashed-names? true
+                                                    :quoting :mysql)))]
+        (-> (apply h/select (map last (:select honeysql-query)))
+            (h/from (h/merge-select honeysql-query [(hsql/raw over-clause) :__rownum__]))
+            (h/where [:> :__rownum__ offset])
+            (h/limit items))))))
+
+;; lots of copy-paste here. consider making some of those non-private instead.
+(defn- exception->nice-error-message ^String [^java.sql.SQLException e]
+  (or (->> (.getMessage e)     ; error message comes back like 'Column "ZID" not found; SQL statement: ... [error-code]' sometimes
+           (re-find #"^(.*);") ; the user already knows the SQL, and error code is meaningless
+           second)             ; so just return the part of the exception that is relevant
+      (.getMessage e)))
+
+(defn do-with-try-catch {:style/indent 0} [f]
+  (try (f)
+       (catch java.sql.SQLException e
+         (log/error (jdbc/print-sql-exception-chain e))
+         (throw (Exception. (exception->nice-error-message e))))))
```

Comment:
This looks like an exact copy-paste of the code in `metabase.driver.generic-sql.query-processor`, any reason why you can't just use that implementation?

Reply:
Done (I made it public)